### PR TITLE
prevent _ready() hang when using dynamic agent setup

### LINKED ISSUE #55: `sync.gd` Stuck at `_ready()` When Using Dynamic Agent Setup
I'm working on a project where I reuse the same `EvaluationMap` for both game testing and AI agent training. Due to the complexity of the setup, I dynamically spawn game elements (such as props, flags, weapons, and agents, ...) after the map has been loaded, placing them at random positions, change properties, ... in code.

However, when I instantiate `sync.gd`, it gets stuck at the following line in older versions:

```gdscript
await get_tree().root.ready
```

In the current `main` branch, it now gets stuck at:

```gdscript
await get_parent().ready
```

This happens because the scene has already been fully set up and the `ready` signal has already fired by the time `sync.gd` is instantiated.

### Proposed Fix

Instead of awaiting the `ready` signal, this issue can be resolved by using `call_deferred()` to delay the call to `_initialize()` until the next idle frame—after the current scene tree update is complete. Deferring the initialization ensures that all nodes (including dynamically added agents and dependencies) have finished their own `_ready()` methods, and the parent of this node is also fully initialized.

This guarantees that when `_initialize()` runs, all agents and required nodes are present and ready.

### PR DESCRIPTION
**Pull request by nongvantinh**
Fixes: GH-55

Avoids waiting on already-fired `ready` signals by deferring _initialize() using `call_deferred()`. This ensures all dynamically added nodes and dependencies have completed their own _ready() methods before initialization.

### CODE REVIEW

**Review by Ivan-267 — COMMENTED** [review]
Hello and thanks for this PR, I haven't tried it but the solution seems elegant. I'd like to do a brief test if I get some time on both Windows and Linux, but likely it will be fine.

There might be an edge case if some of the important nodes use "await frame/physics frame/etc" in their ready methods (or are otherwise not loaded within a single frame), but this case can also happen with the previous code.

For all such cases, there might be a simple alternate solution. One possible way might be to extend `sync.gd` in the project, override `_ready()` to do nothing (just return), then just call sync `_initialize()` from your game/scene manager node once the scene has fully loaded.

File: addons/godot_rl_agents/sync.gd [hunk 1]
```diff
@@ -60,11 +60,11 @@ var _obs_space_training: Array[Dictionary] = []
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
-	await get_parent().ready
-	get_tree().set_pause(true)
-	_initialize()
-	await get_tree().create_timer(1.0).timeout
-	get_tree().set_pause(false)
+	call_deferred("_try_initialize")
```
Comment (alpozcan) [inline addons/godot_rl_agents/sync.gd:63]:
```suggestion
	try_initialize.call_deferred()
```
